components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/common/APIMgtLatencyStatsHandler.java

```diff
@@ -65,4 +79,30 @@ public boolean handleResponse(MessageContext messageContext) {
         return true;
     }
 
+    private void setSwaggerToMessageContext(MessageContext messageContext) {
+        // Read OpenAPI from local entry
+        if (openAPI == null && apiUUID != null) {
+            synchronized (this) {
+                if (openAPI == null) {
+                    long startTime = System.currentTimeMillis();
+                    Entry localEntryObj = (Entry) messageContext.getConfiguration().getLocalRegistry().get(apiUUID);
+                    if (localEntryObj != null) {
+                        swagger = localEntryObj.getValue().toString();
+                        OpenAPIParser parser = new OpenAPIParser();
+                        openAPI = parser.readContents(swagger,
+                                null, null).getOpenAPI();
+                    }
+                    long endTime = System.currentTimeMillis();
+                    if (log.isDebugEnabled()) {
+                        log.debug("Time to parse the swagger(ms) : " + (endTime - startTime));
+                    }
+                }
+            }
+        }
+        // Add OpenAPI to message context
+        messageContext.setProperty(APIMgtGatewayConstants.API_SWAGGER, openAPI);
+        // Add swagger String to message context
+        messageContext.setProperty(APIMgtGatewayConstants.SWAGGER, swagger);
```

Comment:
shall we use OpenAPIObject and OpenAPIString here?

Reply:
renamed as suggested.